src/Orchard.Cms.Web/Modules/Orchard.OpenId/Drivers/OpenIdSettingsDisplayDriver.cs

```diff
@@ -79,7 +97,13 @@ public override IDisplayResult Edit(OpenIdSettings settings, BuildEditorContext
                 settings.AllowImplicitFlow = model.AllowImplicitFlow;
                 settings.AllowHybridFlow = model.AllowHybridFlow;
 
-                _openIdServices.IsValidOpenIdSettings(settings, updater.ModelState);
+                
+                if (_openIdServices.IsValidOpenIdSettings(settings, updater.ModelState) && _memoryCache.Get(RestartPendingCacheKey) == null)
+                {
+                    var entry = _memoryCache.CreateEntry(RestartPendingCacheKey);
+                    entry.SetPriority(CacheItemPriority.NeverRemove);
+                    _memoryCache.Set(entry.Key, entry);
```

Comment:
Isn't there an overload which takes a priority directly (author of these extensions speaking)